docs/rfc/rfc-7.adoc

```diff
@@ -0,0 +1,220 @@
+transaction fee reflecting the current situation in the Bitcoin network. To
+leave enough time for retries, deposits for which Bitcoin transactions unlock
+for a refund earlier than two weeks from now are skipped.
+
+The Coordinator must not propose a sweep if there is currently a pending signing
+operation being executed by the wallet or if the last signed transaction does
+not have 6 Bitcoin confirmations yet.
+
+The Coordinator proposes a sweep over the wallet's broadcast channel.
+
+All signing group members must confirm that all deposits in the proposed sweep
+are sufficiently confirmed, that the maximum number of deposits is not exceeded,
+that the proposed fee reflects the current situation in the Bitcoin network
+plus/minus some margin, that all deposit UTXOs can be unlocked with the
+wallet's public key, and that there is enough time to unlock UTXO before the
+refund.
+
+If all these requirements are met, the last transaction executed by the wallet
+is sufficiently confirmed, and there is no other signing by the wallet in
+progress, the signing group members proceed with tECDSA signing protocol. Each
+input is signed in a separate state machine instance, independently from other
+signatures being executed.
+
+
+==== Retries
+
+The signing protocol should be extended by an additional phase at the very
+beginning of each retry that will inform about the data being signed. For
+sweeping, it will inform about the Bitcoin sweep transaction proposed by the
+Coordinator. All signing members should broadcast this information if they have
+already validated the proposition. The signing group members continue with the
+protocol only if there are at least 51 messages informing them about the
+transaction and its successful validation. Group members who have turned on
+their nodes can learn about the transaction the rest of the group wants to sign
+and they can join the signing protocol if necessary.
+
+The signing protocol takes 72 blocks. Assuming an average block time of 12
+seconds, it takes approximately 15 minutes to execute the protocol.
+
+- With 2 malicious members in a signing group, we need 5 retries of the protocol
+in the worst case (`P = (98 choose 51) / (100 choose 51) = 0.23757575757`),
+which takes 1h 15min.
+- With 3 malicious members in a signing group, we need 10 retries of the
+protocol in the worst case (`P = (97 choose 51) / (100 choose 51) = 0.11393939393`),
+which takes 2h 30 min.
+- With 4 malicious members in a signing group, we need 20 retries of the
+protocol in the worst case (`P = (96 choose 51) / (100 choose 51) = 0.05403311465`),
+which takes 5h.
+- With 5 malicious members in a signing group, we need 50 retries of the
+protocol in the worst case (`P = (95 choose 51) / (100 choose 51) = 0.02532802249`),
+which takes 12h 30 min.
+
+Given the fact, the first few retries will focus on inactivity and the
+probability of having more than 2 malicious members in a signing group is low,
```

Comment:
I don't think this is true (https://github.com/keep-network/keep-core/pull/3352#issuecomment-1274670323). As we discussed, we plan to use the announcement phase before the signature attempt to filter out inactive members early.

Reply:
Addressed in the most recent version.